### PR TITLE
Add oauth domain to CBS

### PR DESCRIPTION
**Pull request by Wawrzyn321**
**Description**

Changes proposed in this pull request:

- Add OAuth2Clients in `oauth` domain

**Testing GQL**
```
fragment Client on OAuth2Client {
  name
  namespace
  generation
  error {
    code
    description
  }
  spec {
    grantTypes
    responseTypes
    scope
    secretName
  }
}

query Rs {
  oAuth2Clients(namespace: "default") {
    ...Client
  }
}

query R {
  oAuth2Client(name: "another-name", namespace: "default") {
    ...Client
  }
}

mutation C {
  createOAuth2Client(name: "test", namespace: "default", params: {
    grantTypes: ["client_credentials"]
    responseTypes: ["id_token", "code", "token"]
    scope: "read write"
    secretName: "tets"
  }) {
    ...Client
  }
}

mutation U {
  updateOAuth2Client(name: "client-name", namespace: "default", generation:2, params: {
    grantTypes: ["authorization_code"]
    responseTypes: ["id_token", "token"]
    scope: "read write speak"
    secretName: "namee"
  }) {
    ...Client
  }
}

mutation D {
  deleteOAuth2Client(name: "client-name-2", namespace: "default") {
    ...Client
  }
}

subscription S {
  oAuth2ClientEvent(namespace: "default") {
    type
    client {
      # ...Client
      name
    }
  }
}
```

**Related issue(s)**
[1891](https://github.com/kyma-project/console/issues/1891)

### CODE REVIEW

File: components/console-backend-service/internal/domain/oauth/oauth2client.go [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+import (
+	"github.com/kyma-project/kyma/components/console-backend-service/internal/resource"
+)
+
+type Resolver struct {
+	*resource.Module
+}
+
+func New(factory *resource.GenericServiceFactory) *Resolver {
+	module := resource.NewModule("apigateway", factory, resource.ServiceCreators{
```
Comment (akucharska) [inline components/console-backend-service/internal/domain/oauth/oauth2client.go:12]: Shouldn't this be "oauth"

Comment (Wawrzyn321) [reply components/console-backend-service/internal/domain/oauth/oauth2client.go:12]: It was originally (as changed in `Reuse BackendModule` commit), but if `oauth` and `apigateway` were to be defined always together, it makes sense to use the same module.

**Pull request merged**